lib/sqlalchemy/sql/elements.py

```diff
@@ -3836,7 +3926,7 @@ def __init__(self, start, stop, step, _name=None):
         )
         self.type = type_api.NULLTYPE
 
-    def self_group(self, against=None):
+    def self_group(self, against: Optional[OperatorType] = None) -> Slice:
         assert against is operator.getitem
```

Comment:
Getting a non-overlapping identity check. `operator.getitem()` is interpreted as an overloaded function by mypy. Hovewer the overloads it sees are from `typeshed-fallback/stdlib/_operator.pyi` (from inside `pylance`) stub. I assume this should not work this way. Found something on `typeshed` [here](https://github.com/microsoft/pylance-release/issues/840), will try to find a solution.